### DISCUSSION TITLE
Game randomly changes audio offset

### DISCUSSION THREAD

**Discussion by Robi0990** [ppy/osu#27567] [Q&A] [answered]
### Type

Game behaviour

### Bug description

The game at random times changes audio offset by a random amount (I've encountered from ~+20ms to ~+50ms). I've attached a video of me getting the bug and it seems to be reproducible. (I've also had it sometimes happen when I was alt-tabbed out of the game for a while which gave me a ~+50ms offset)

### Screenshots or videos

https://github.com/ppy/osu/assets/67930895/c9613c69-93ca-4481-b3b4-cbcbc39f0783

### Version

2024.302.1

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/14552667/compressed-logs.zip)

**Comment by peppy** [comment]
Can you explain what you're showing in the video? It's not immediately clear.

**Reply by Robi0990** [reply]
At the end of the video, the drum kicks when hovering on the osu! logo are more delayed when compared to the beginning of the video. It can be heard easier if the video is slowed down a bit. This affects gameplay as well, the notes appear sooner compared to the audio.

**Reply by Robi0990** [reply]
https://github.com/ppy/osu/assets/67930895/4bdd8036-0dcf-438b-aa56-f0ac043bc14b

Here's the bug when in gameplay. I show in the video that the map offset and universal offset never change and the recommended offset after the map showed 20ms. After a restart, it goes to 3ms (basically 0ms) without changing anything.

**Comment by peppy** [comment]
If you're still encountering this, please read https://github.com/ppy/osu/issues/31843 and test the provided fix if you're able!

**Reply by Robi0990** [reply, marked as answer]
This got fixed for me a while ago, don't know if it was a lazer update that fixed it or something else. My issue started around when audio offset suggestion was introduced (version 2023.1229.0) in the game if it means anything.
